visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -100,115 +104,227 @@ export class NodeDecorator {
+						name,
+						EdgeAttributeType.INCOMING
+					)
+					setEdgeMediansToParent(
+						medians,
+						`${MedianSelectors.OUTGOING}${selector}`,
+						`${MedianSelectors.OUTGOING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.OUTGOING
+					)
+					collectEdgeMediansOnParent(
+						medians,
+						`${MedianSelectors.INCOMING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.INCOMING
+					)
+					collectEdgeMediansOnParent(
+						medians,
+						`${MedianSelectors.OUTGOING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.OUTGOING
+					)
+				} else {
+					parent.data.edgeAttributes[name].incoming += value.incoming
+					parent.data.edgeAttributes[name].outgoing += value.outgoing
 				}
 			}
 		})
-		return map
-	}
 
-	private static aggregateLeafMetric(metrics: KeyValuePair[], metricName: string, attributeTypes: AttributeTypes) {
-		if (attributeTypes.nodes[metricName] === AttributeTypeValue.relative) {
-			const metricValues: number[] = []
-			for (const metric of metrics) {
-				const number = metric[metricName]
-				if (number) {
-					metricValues.push(number)
+		// Set Median for root node
+		for (const name of edgeKeys) {
+			if (attributeTypes.edges[name] === AttributeTypeValue.relative) {
+				map.edgeAttributes[name].incoming = getMedian(medians.get(`${MedianSelectors.INCOMING}${name}${map.path}`))
+				map.edgeAttributes[name].outgoing = getMedian(medians.get(`${MedianSelectors.OUTGOING}${name}${map.path}`))
+			}
+		}
+
+		for (const name of attributeKeys) {
+			if (attributeTypes.nodes[name] === AttributeTypeValue.relative) {
+				map.attributes[name] = getMedian(medians.get(`${MedianSelectors.MEDIAN}${name}${map.path}`))
+				if (isDeltaState) {
+					map.deltas[name] = getMedian(medians.get(`${MedianSelectors.DELTA}${name}${map.path}`))
 				}
 			}
+		}
+	}
+}
+
+function collectEdgeMediansOnParent(
+	medians: Map<string, number[]>,
+	selector: string,
+	child: CodeMapNode,
+	metricName: string,
+	type: EdgeAttributeType
+) {
+	if (child.edgeAttributes[metricName][type] !== 0) {
+		collectMedians(medians, selector, child, child.edgeAttributes[metricName][type])
+	}
+}
+
+function collectNodeMediansOnParent(
+	medians: Map<string, number[]>,
+	parentSelector: string,
+	child: CodeMapNode,
+	metricName: string,
+	isDeltaState: boolean
+) {
+	if (child.attributes[metricName] !== 0) {
+		collectMedians(medians, `${MedianSelectors.MEDIAN}${parentSelector}`, child, child[NodeAttributeType.ATTRIBUTES][metricName])
+	}
 
-			if (metricValues.length === 0) {
-				return 0
+	if (isDeltaState && child.deltas[metricName] !== 0) {
+		collectMedians(medians, `${MedianSelectors.DELTA}${parentSelector}`, child, child[NodeAttributeType.DELTAS][metricName])
+	}
+}
+
+function setNodeMediansToParent(
+	medians: Map<string, number[]>,
+	selector: string,
+	parentSelector: string,
+	child: CodeMapNode,
+	metricName: string,
+	isDeltaState: boolean
+) {
+	if (!isLeaf(child)) {
```

Comment:
I suggest to invert the condition to prevent deep indentations: `if (isLeaf(child)) { return }`